content/docs/tutorials/understanding_metric_types.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Understanding metric types  
+sort_rank: 2
+---
+
+# Types of metrics.
+
+Prometheus supports four types of metrics, they are
+    - Counter
+    - Gauge
+    - Histogram
+    - Summary 
+
+## Counter
+
+Counter is a metric value which can only increase or reset i.e the value cannot reduce than the previous value. It can be used for metrics like number of requests, no of errors etc.
+
+Type the below query in the query bar and click execute.
+
+<code>go\_gc\_duration\_seconds\_count</code>
+
+
+[![Counter](/assets/tutorial/counter_example.png)](/assets/tutorial/counter_example.png)
+
+The rate() function in PromQL takes the history of metrics over a time frame and calculates how fast value is increasing per second. Rate is applicable on counter values only.
+
+<code> rate(go\_gc\_duration\_seconds\_count[5m])</code>
+[![Rate Counter](/assets/tutorial/rate_example.png)](/assets/tutorial/rate_example.png)
+
+## Gauge
+
+Gauge is a number which can either go up or down. It can be used for metrics like number of pods in a cluster, number of events in an queue etc.
+
+<code> go\_memstats\_heap\_alloc\_bytes</code>
+[![Gauge](/assets/tutorial/gauge_example.png)](/assets/tutorial/gauge_example.png)
+
+PromQL functions like `max_over_time`, `min_over_time` and `avg_over_time` can be used on gauge metrics
+
+## Histogram
+
+Histogram is bit complex metric type when compared to the previous two. Histogram can be used for any calculated value which is counted based on bucket values, the buckets can be configured by the user. It is a cumulative metric and provides sum of all the values by default. This is applicable for metrics like request time, cpu temperature etc
+
+Example: Lets sat we want to observe the time taken to process api requests. Instead of storing the request time for each request, histogram metric allows us to approximate and store frequency of requests which fall into particular buckets. We define buckets for time taken like 0.3,0.5,0.7,1,1.2. So these are our buckets and once the time taken for a request is calculated it is added to the count of all the buckets that are higher than the value.
```

Comment:
```suggestion
Example: Lets sat we want to observe the time taken to process api requests. Instead of storing the request time for each request, histograms allow us to store them in buckets. We define buckets for time taken, for example `lower or equal 0.3` , `le 0.5`, `le 0.7`, `le 1`, and `le 1.2`. So these are our buckets and once the time taken for a request is calculated it is added to the count of all the buckets which bucket boundaries are higher than the measured value.
```